dipy/nn/finta_model.py

```diff
@@ -0,0 +1,435 @@
+import logging
+
+import numpy as np
+import nibabel as nib
+
+from dipy.nn.utils import pre_pad
+from dipy.nn.utils import dict_kernel_size_flatten_encoder_shape
+
+from dipy.testing.decorators import doctest_skip_parser, warning_for_keywords
+from dipy.utils.optpkg import optional_package
+
+
+tf, have_tf, _ = optional_package("tensorflow", min_version="2.0.0")
+if have_tf:
+    import tensorflow as tf
+    import keras
+    from keras import layers, Model, initializers
+    from keras.layers import Layer
+else:
+    class Model:
+        pass
+
+    class Layer:
+        pass
+
+    logging.warning(
+        "This model requires Tensorflow.\
+                    Please install these packages using \
+                    pip. If using mac, please refer to this \
+                    link for installation. \
+                    https://github.com/apple/tensorflow_macos"
+    )
+
+logging.basicConfig()
+
+class Encoder(Layer):
+    def __init__(self, latent_space_dims=32, kernel_size=3, **kwargs):
+        """
```

Comment:
Start the docstring immediately after the triple quotes. Applies to all docstrings.

Reply:
Done